examples/docs_projects/project_components_pdf_extraction/pdf_extraction/lib/pdf_extraction.py

```diff
@@ -0,0 +1,273 @@
+import os
+from dataclasses import dataclass
+from pathlib import Path
+
+import dagster as dg
+from dagster.components import (
+    Component,
+    ComponentLoadContext,
+    Resolvable,
+    Scaffolder,
+    ScaffoldRequest,
+)
+
+from .pdf_extraction_resource import PDFTextExtractor
+
+
+@dataclass
+class PdfExtraction(Component, Resolvable):
+    """A component for extracting and validating text from PDF documents.
+
+    This component provides a complete PDF text extraction pipeline that:
+    1. Converts PDF pages to high-quality images
+    2. Performs OCR text extraction using Tesseract
+    3. Validates extraction quality using OpenAI
+    4. Generates detailed extraction reports and metrics
+
+    The component creates three main assets:
+    - {pdf_name}_convert_to_image: Converts PDF pages to images
+    - {pdf_name}_extract_text: Extracts text from the images using OCR
+    - Asset check on extract_text: Validates extraction quality using OpenAI
+
+    Configuration:
+        pdf_path (str): Path to the PDF document to process
+        output_dir (str): Directory for storing extracted images and text
+        asset_specs (Sequence[ResolvedAssetSpec]): Asset specifications for the pipeline
+        language (str): OCR language code (e.g., 'eng' for English)
+        dpi (int): DPI resolution for PDF to image conversion
+        openai_api_key (str): OpenAI API key for validation
+        openai_model (str): OpenAI model to use for validation
+        validation_score (int): Minimum acceptable validation score (1-10)
+    """
+
+    pdf_dir: str
```

Comment:
This is me not knowing components. But defaults need to be set in this class. It seems like the same defaults are also set in the `PdfExtractionScaffolder`